storage/storage_src.go

```diff
@@ -129,7 +130,13 @@ func (s *storageImageSource) GetBlob(ctx context.Context, info types.BlobInfo, c
 	if err != nil {
 		return nil, 0, err
 	}
-
+	// On Unix we can eagerly unlink the file to ensure it's automatically
+	// cleaned up on process termination (or if the caller forgets to invoke Close())
+	if runtime.GOOS != "windows" {
+		if err := os.Remove(tmpFile.Name()); err != nil {
+			return nil, 0, err
+		}
+	}
 	if _, err := io.Copy(tmpFile, rc); err != nil {
```

Comment:
Looking at the code… these error paths are buggy, they should be deleting the file.

Could you fix that as well, please? Something very vaguely like
```go
tmpFile := …
readCloser := tmpFile
if !windows { os.Remove(…) }
else {
    readCloser = NewReadCloserWrapper(…)
}
defer func() {
    if !succeeded { readCloser.Close() }
}
```

would fix that, _and_ move the “ensure the file is eventually removed” logic into one place instead of two as it is now.

(We could even extract that “create a temporary file and ensure it is removed by `Close` at the latest” into a separate utility subpackage, but that can easily wait until we have more users.)

Reply:
Done.

This code is IMO a perfect example of the "superficial simplicity" of Go materializing as bugs.  In Rust, ownership semantics (instead of potentially arbitrarily delayed GC) would ensure the file is closed by default (in `Drop`) reliably.  Second, conditional compilation is *much* nicer with things like `if cfg!()` - splitting the code into separate files is a big hammer here, and so here I just chose to dynamically dispatch at runtime on the platform, which one finds even in the Go standard library.

This is very familiar to me because I did https://github.com/bytecodealliance/cap-std/commit/a521444939b35b7ef30bdc445b7381fc7ee58a95

(Hmm though now that I think about it, I guess `new_anonymous()` likely doesn't work on Windows...)

Reply:
> "superficial simplicity" of Go

Oh yes.